src/components/intent.ts

```diff
@@ -41,6 +45,12 @@ export interface IntentOpts {
     dontJoin?: boolean;
     enablePresence?: boolean;
     registered?: boolean;
+    encryption?: {
+        sessionPromise: Promise<ClientEncryptionSession|null>;
+        sessionCreatedCallback: (session: ClientEncryptionSession) => Promise<void>;
+        ensureSyncingCallback: () => Promise<void>;
```

Comment:
Variable naming: Having looked at all occurrences, I'm not really sure what `ensureSyncingCallback` means.

Reply:
I've renamed it to ensureClientSyncingCallback. Essentially, it ensures the intent's client is /syncing before it performs any encrypted actions.